File: python/stdlib/structures/whilegreater.py
try:
    from ...stdlib import Node as base
except ValueError:
    from stdlib import Node as base


class Node(base.Node):
    def __init__(self, verbose, args):
        super(Node, self).__init__("While greater", "structures.whilegreater", "",
                                   {"initial":"Number", "val":"Number"},
                                   {"loop": "Number", "leave":"Number"},
                                   "Loop while initial is greater than val.", verbose)

    def tick(self, value):
        if value["initial"] > value["val"]:
            return {"loop": value["initial"], "leave": None}
        else:
            return {"loop": None, "leave": value["initial"]}

File: lua/stdlib/structures/whilegreater.lua
local myNode = {}
function myNode.init(node)
  node.tick = function(value)
    result = {}
    result.loop = nil
    result.leave = nil
    if value.initial > value.val then
      result.loop = value.val
    else
      result.leave = value.val
    end
    return result
  end
end

function myNode.spec(node)
  node.name = "While greater"
  node.inputs.val = "Number"
  node.inputs.initial = "Number"
  node.outputs.loop = "Number"
  node.outputs.leave = "Number"
  node.desc = "Loop while initial is greater than val."
end

return myNode
